Incorrectly parsed XPath locator with nested functions

**Describe the bug**
'SyntaxError' when trying to use XPath locator in any step with the next pattern (sequence):
1. `contains` function
2. any nested function in `contains`
3. not `AND` or `OR` operators
4. any function

Error:
``  org.vividus.softassert.exception.VerificationError:  java.lang.AssertionError: Potentially known issue: EVA-01 (Type: INTERNAL. Status: UNKNOWN). Wait with timeout of 20 seconds. Condition: org.vividus.ui.action.search.AbstractElementAction$1@3e7cb979. Error: invalid selector: Unable to locate an element with the xpath expression //*[contains(normalize-space(text(), 'night')]/parent::body/label[contains(text()), 'story')] because of the following error:
SyntaxError: Failed to execute 'evaluate' on 'Document': The string '//*[contains(normalize-space(text(), 'night')]/parent::body/label[contains(text()), 'story')]' is not a valid XPath expression.
  (Session info: chrome=110.0.5481.96) ``

Original XPath in chrome:
`//*[contains(text(), 'night')]/parent::body/label[contains(text(), 'story')]`
![image](https://user-images.githubusercontent.com/58571912/219388486-22e4ab08-d008-45b5-a20c-a57e1d3ab233.png)

**To Reproduce**
Run the next step on any page:
`` Then number of elements found by `xpath(//*[contains(text(), 'night')]/parent::body/label[contains(text(), 'story')])` is equal to `1`  ``

**Artifacts**
[allureXpathNormalization.zip](https://github.com/vividus-framework/vividus/files/10756844/allureXpathNormalization.zip)

**Execution environment (please complete the following information):**
 - Vividus version: 0.5.4

--- a/vividus-extension-selenium/src/main/java/org/vividus/ui/util/XpathLocatorUtils.java
+++ b/vividus-extension-selenium/src/main/java/org/vividus/ui/util/XpathLocatorUtils.java
@@ -35,7 +35,7 @@ public final class XpathLocatorUtils
     private static final Pattern ANY_ATTR_OR_TEXT_VALUE_PATTERN = Pattern.compile("((@[*])|(text\\(\\)))(| )="
             + CONCAT);
     private static final Pattern CONTAINS_PATTERN = Pattern
-            .compile("contains\\((?!normalize-space)(([a-z-]+?)\\(((?!\\b(and|or)\\b).)*\\)|.+?),");
+            .compile("contains\\((?!normalize-space)(([a-z-]+?)\\(((?!(and|or|])).)*\\)|.+?),");
     private static final String NORMALIZE_SPACE_FORMAT = "$3normalize-space($4$5$6)=$9";
     private static final String ANY_ATTR_OR_TEXT_NORMALIZE_SPACE_FORMAT = "$1[normalize-space()=$6]";
     private static final String NORMALIZE_SPACE_TRANSLATE_FORMAT = "normalize-space($1)=$5$6$7";

--- a/vividus-extension-selenium/src/test/java/org/vividus/ui/util/XpathLocatorUtilsTests.java
+++ b/vividus-extension-selenium/src/test/java/org/vividus/ui/util/XpathLocatorUtilsTests.java
@@ -86,14 +86,17 @@ class XpathLocatorUtilsTests
     }
 
     @Test
-    void testGetXpathPatternWithNormalizationAndNestedFunctionsInsideContains()
+    void testGetXpathPatternWithNormalizationAndSequenceOfNestedFunctionsWithContains()
     {
         By expectedLocator = By.xpath("//*[contains(normalize-space(translate(@class, \"abcdefghijklmnopqrstuvwxyz\", "
                 + "\"ABCDEFGHIJKLMNOPQRSTUVWXYZ\")), 'MAIN') "
                 + "or contains(normalize-space(translate(concat(text(), 'page'), \"abcdefghijklmnopqrstuvwxyz\", "
-                + "\"ABCDEFGHIJKLMNOPQRSTUVWXYZ\")), 'MAIN PAGE')]");
+                + "\"ABCDEFGHIJKLMNOPQRSTUVWXYZ\")), 'MAIN PAGE')]"
+                + "//*[contains(normalize-space(text()), 'Tachi')]"
+                + "/parent::body/label[contains(normalize-space(text()), 'MCRN')]");
         By actualLocator = XpathLocatorUtils.getXPathLocator("//*[contains(translate(@class, %1$s, %2$s), 'MAIN') "
-                + "or contains(translate(concat(text(), 'page'), %1$s, %2$s), 'MAIN PAGE')]",
+                + "or contains(translate(concat(text(), 'page'), %1$s, %2$s), 'MAIN PAGE')]"
+                        + "//*[contains(text(), 'Tachi')]/parent::body/label[contains(text(), 'MCRN')]",
                 "abcdefghijklmnopqrstuvwxyz", "ABCDEFGHIJKLMNOPQRSTUVWXYZ");
         assertEquals(expectedLocator, actualLocator);
     }

